### PR TITLE
Editorial comments on draft-12

### PR DESCRIPTION
**Pull request by timbray**

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -1501,8 +1505,7 @@ immediate contexts, which lead to the following conditions for well-typedness:
 
 {:vspace}
 As a `test-expr` in a logical expression:
-: The function's declared result type is `LogicalType`, or
-  (giving rise to conversion as per {{type-conv}}) `NodesType`.
+: The function's declared result type is `LogicalType`.
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1508]: This change would mean a function with declared type NodesType is no longer well-typed here.
I think Glyn is right that this section needs to be consistent on its own.

File: draft-ietf-jsonpath-base.md [hunk 2]
```diff
@@ -1301,7 +1301,7 @@ the comparison is between values which are both numbers or both strings and whic
       the remainder of the first string compares less than the remainder of the second string.
 
 Note that comparisons using the operator `<` yield false if either value being
-compared is an object, array, boolean, or `null`.
+compared is an object, array, boolean, `null`, or (see below) `Nothing`.
```
Comment (glyn) [inline draft-ietf-jsonpath-base.md:1304]: I've cherry-picked this into https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/454 and tried to improve it there.

File: draft-ietf-jsonpath-base.md [hunk 3]
```diff
@@ -1367,8 +1367,12 @@ Sections {{<match}} and {{<search}} below for details about these.)
...
 
     {
-      "a": [3, 5, 1, 2, 4, 6, {"b": "j"}, {"b": "k"},
-            {"b": {}}, {"b": "kilo"}],
+      "a": [3, 5, 1, 2, 4, 6,
+            {"b": "j"},
+            {"b": "k"},
+            {"b": {}},
+            {"b": "kilo"}
+           ],
```
Comment (glyn) [inline draft-ietf-jsonpath-base.md:1375]: I cherry-picked the above improvement into https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/455.

File: draft-ietf-jsonpath-base.md [hunk 4]
```diff
@@ -1479,9 +1483,9 @@ parameter of declared type `LogicalType`, with the equivalent conversion rule:
   * If the nodelist contains one or more nodes, the conversion result is `LogicalTrue`.
   * If the nodelist is empty, the conversion result is `LogicalFalse`.
 
-Extraction of a value from a nodelist can be performed in several
-ways, so an implicit conversion from `NodesType` to `ValueType`
-may be surprising and has therefore not been defined.
```
Comment (glyn) [inline draft-ietf-jsonpath-base.md:1484]: I've attempted a rewording of this in https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/456.

**Comment by glyn** [comment]
@timbray I think we've merged what we can. Anything from here you're still keen to see?

**Comment by timbray** [comment]
I think everything I suggested has been looked at.

**Comment by glyn** [comment]
Ok, thanks.

**Pull request closed without merging**